subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -597,6 +597,25 @@ public int compareTo(Project otherProject) {
         return ProjectOrderingUtil.compare(this, otherProject);
     }
 
+    @Override
+    public int hashCode() {
+        return super.hashCode();
```

Comment:
❓  This looks suspicious because it does not delegate hash code computation to the identity path.

Reply:
Strictly speaking, in my understanding it's not really matters what the `hashCode` returns here.
The contract `equals <-> hashCode`: if `a.equals(b) == true` => `a.hashCode == b.hashCode`. 
Given that identityPath is being checked in `DefaultProject#equals` implementation, it's effectively means that each project is unique and only equal to itself. Wrappers are delegating hashCode calculation to the delegate. 
Hence, `hashCode` implementation not necessary must to be computed from the same entity(`identityPath`, in our case) that used for equality logic.

Maybe I miss something, tho

Reply:
Darn it, GitHub not showing the responses to comments...

I guess your conclusion is formally correct, but it is entirely dependent on the wrappers always delegating the hash code. Even if we don't change that logic, I find it hard to reason about and maintain.

I'd think about equals and hashCode here as a "trait" being implemented by delegation. They should delegate to the same thing.